enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,349 @@
+
+	// MinSize is the minimum size of the device which needs to be included
+	// +optional
+	MinSize *resource.Quantity `json:"minSize"`
+
+	// MaxSize is the maximum size of the device which needs to be included
+	// +optional
+	MaxSize *resource.Quantity `json:"maxSize"`
+
+	// Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Models []string `json:"models"`
+
+	// Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+	// to contain at least one of these strings.
+	// +optional
+	Vendors []string `json:"vendors"`
+}
+
+type LocalVolumeGroupPhase string
+
+const (
+	DiscoveringPhase LocalVolumeGroupPhase = "Discovering"
+	FailedPhase      LocalVolumeGroupPhase = "Failed"
+	DiscoveredPhase  Phase                 = "Discovered"
+	ProvisionedPhase LocalVolumeGroupPhase = "Provisioned"
+)
+
+type LocalVolumeGroupStatus struct {
+	// Phase describes the state of the LocalVolumeGroup
+	Phase LocalVolumeGroupPhase `json:"phase,omitempty"`
+
+	// A human-readable message indicating details about why the LocalVolumeGroup is in this state.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// Reason is a brief CamelCase string that describes any failure and is meant
+	// for machine parsing and tidy display in the CLI.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// TotalDiscoveredDeviceCount is the count of the total devices which matched the inclusion filter
+	TotalDiscoveredDeviceCount *int32 `json:"totalDiscoveredDeviceCount,omitempty"`
+
+	// TotalProvisionedDeviceCount is the count of the total devices over which the PVs has been provisioned
+	TotalProvisionedDeviceCount *int32 `json:"totalProvisionedDeviceCount,omitempty"`
+
+  // LastProvisionedTimeStamp is the timeStamp value for lastProvisionedTimeStamp
+  // +optional
+	LastProvisionedTimeStamp `json:"lastProvisionedTimeStamp,omitempty"`
+
+	// observedGeneration is the last generation change the operator has dealt with
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+```
+### Workflow of LocalPV creation via LocalVolumeGroup CR
+- Once the `LocalVolumeGroup` is created, the controller will:
+  - configure the local-static-provisioner to make a new StorageClass based on certain directories on the selected nodes.
+  - assign diskmaker daemons to the selected nodes.
+- The diskmaker daemon will find devices that match the disovery policy and symlink them into the directory that the local-static-provisioner is watching.
+- Possible `status.phase` values for `LocalVolumeGroup`
+  - `Discovering`
+  - `Failed`
+  - `Discovered`
+  - `Provisioned`
+
+#### Note: There is a chance of race condition: 
+- If two `LocalVolumeGroup` CR targets same nodes with overlapping inclusion filter.
+- If `LocalVolumeGroup` CR and `LocalVolume` CR targets the same node with overlapping devices.
```

Comment:
If there a `LocalVolumeGroup` object created that targets a specific node and then user creates a simple `LocalVolume` object that targets same node, will this result in creation of 2 different daemonsets that target same node?

Reply:
Nope, the plan is to use the existing daemonset with modification. So if a daemonset already exist s on a node we will be using the same.

Reply:
yeah makes sense

Reply:
discussed this offline with @openshift/sig-storage team and we agreed that, this is a fix that we have to do. This is out of scope for this enhancement but having a single pod on node managing all local-volumes will prevent weird race conditions. 

I will work on a follow up.